### PR TITLE
user: Fix flaky locale test and change locale on setting update

### PR DESCRIPTION
**Pull request by david-yz-liu**
## Motivation and Context

There is currently a flaky test in `user_controllers_spec.rb:65` (`I18n.locale` wasn't being set properly). While fixing this, I noticed a small bug in the `UsersController#update_settings` method, which didn't change `I18n.locale` even when the user changed this setting.

## Your Changes

**Description**: Fixed the test setup and modified `UsersController#update_settings` to change `I18n.locale`.

**Type of change** (select all that apply):

 

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Test update (change that modifies or updates tests only)

## Testing

Ran the tests locally a few times, always passed. Checked in the UI that after changing the locale setting, the flash message that's rendered is in the updated locale, rather than the previous locale

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have fixed any Hound bot comments. 
- [x] I have verified that the TravisCI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -150,21 +146,17 @@
...
+      before :each do
         I18n.locale = :en
       end
 
-      it 'updates locale for student' do
+      it 'updates locale' do
         locale = 'es'
-        expect(I18n.locale).to eq I18n.default_locale
-        patch_as student,
+        patch_as admin,
```
Comment (mishaschwartz) [inline spec/controllers/users_controller_spec.rb:155]: `patch_as student` ?

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -26,16 +26,16 @@
     end
 
     describe 'Change locale' do
-      it 'updates locale for admin' do
+      before :each do
```
Comment (mishaschwartz) [inline spec/controllers/users_controller_spec.rb:29]: If you want a more compact syntax:

```ruby
before { I18n.local = :en }
```

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -26,16 +26,16 @@
...
+        I18n.locale = :en
+      end
+
+      it 'updates locale' do
```
Comment (mishaschwartz) [inline spec/controllers/users_controller_spec.rb:33]: This looks like a good candidate for a shared test. Just swap out the `patch_as X` for `patch_as user` and then set the `user` let variable to admin, student, ta.

**Comment by david-yz-liu** [comment]
Note: coverage decreased because the tests are no longer accessing the `check_timeout` route, but that's separate from this PR. We'll add tests for that route separately.

**Pull request merged**